Create service subsets and move routing/route-rule* to networking/virtual-service*

andraxylia opened the pull request:
Related to istio/istio#6940

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.github.io/pull/1715#" title="Author self-approved">andraxylia</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **sdake**

Assign the PR to them by writing `/assign @sdake` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.github.io/blob/master/content/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["sdake"]} -->

geeknoid commented on content/docs/examples/bookinfo/index.md:
```diff
@@ -136,7 +136,7 @@ To start the application, follow the instructions below corresponding to your Is
     {{< /text >}}
 
-#### Determining the ingress IP and port
+#### Determining the ingress gateway IP and port
```
You'll need to fix various links pointing here.

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -20,14 +20,39 @@ This task shows you how to configure dynamic request routing based on weights an
 
 ## Content-based routing
 
+The Bookinfo sample deploys multiple versions of each microservice, so we will start by creating DestinationRules
```
Please avoid using "we". That is, it should be something like "so you will start by"

andraxylia replied:
done

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -113,7 +132,7 @@ you'll need to use `replace` rather than `create` in the following command.
     > The corresponding `subset` definitions can be displayed using `istioctl get destinationrules -o yaml`.
 
-    Since rule propagation to the proxies is asynchronous, you should wait a few seconds for the rules
+    Since rule propagation to the proxies is asynchronous, you should wait a few seconds for the VirtualServices
```
should -> must

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -180,16 +199,22 @@ all users to v2, optionally in a gradual fashion. We'll explore this in a separa
 ## Cleanup
 
-*   Remove the application routing rules.
+*   Remove the application VirtualServices.
```
Do we want bullets here or numbered list?

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
````diff
@@ -20,14 +20,39 @@ This task shows you how to configure dynamic request routing based on weights an
+
+    You can display the DestinationRules that are defined with the following command:
+
+    ```command-output-as-yaml
````
Please use the modern {{< text bash yaml >}} sequence here instead of the backticks.
Also, don't indent the content, it should line up with the {{< text >}}

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
````diff
@@ -20,14 +20,39 @@ This task shows you how to configure dynamic request routing based on weights an
+    $ istioctl create -f @samples/bookinfo/routing/destination-rule-all.yaml@
+    ```
+
+    > In a Kubernetes deployment of Istio, you can replace `istioctl`
````
I don't think this paragraph belongs here, let's just remove it.

istio-testing commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/istio/istio.github.io/pull/1715#" title="Author self-approved">andraxylia</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: **fleeto**, **sdake**

Assign the PR to them by writing `/assign @fleeto @sdake` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[content/OWNERS](https://github.com/istio/istio.github.io/blob/master/content/OWNERS)**
- **[content_zh/OWNERS](https://github.com/istio/istio.github.io/blob/master/content_zh/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["fleeto","sdake"]} -->

andraxylia commented:
@geeknoid addressed the code review comments and corresponding PR istio/istio#6940 is merged.
Let's get this in before people start testing with bookinfo.

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -20,26 +20,42 @@ This task shows you how to configure dynamic request routing based on weights an
+    $ istioctl create -f @samples/bookinfo/routing/destination-rule-all.yaml@
+    {{< /text >}}
+
+    If you enabled `mTLS`, please run the following instead
```
mTLS -> mutual TLS with no backticks

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -20,26 +20,42 @@ This task shows you how to configure dynamic request routing based on weights an
+
+    If you enabled `mTLS`, please run the following instead
+
+    {{< text bash yaml >}}
```
No "yaml" since you're not showing YAML output.

geeknoid commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -20,26 +20,42 @@ This task shows you how to configure dynamic request routing based on weights an
+
+    You can display the DestinationRules that are defined with the following command:
+
+    {{< text bash yaml >}}
```
No yaml, as above.

geeknoid commented:
To fix the spelling errors, you'll need to update the .spelling file, or put the offending words in backticks (which is probably the right thing to do for type names like VirtualService and DestinationRule).

To fix the link errors, you'll need to point the web site to the release-1.0 branch of istio/istio. To do this, change the branch_name value in the data/args.yaml file from 'master' to 'release-1.0'

rshriram commented:
@frankbu

frankbu commented on content/docs/tasks/traffic-management/request-routing/index.md:
```diff
@@ -175,23 +194,29 @@ restrictions in order to take advantage of Istio's L7 routing features. Refer to
 
     {{< text bash >}}
-    $ istioctl delete -f @samples/bookinfo/routing/route-rule-all-v1-mtls.yaml@
+    $ istioctl delete -f @samples/bookinfo/routing/destination-rule-all-mtls.yaml@
```
s/routing/networking/

andraxylia commented:
@geeknoid all checks passing, comments addressed please approve.
